### ISSUE TITLE
Incorrect documentation for saving a graph of entities

### ISSUE THREAD

**Issue by litoj** [dotnet/EntityFramework.Docs#4863]
### Type of issue

Code doesn't work

### Description

I am trying to save a graph just like in the example shown. But there are multiple issues with the example:
- whatever I do, it throws an error when saving the `Post`, because it doesn't have the `BlogId` neccessary to save it
- providing the Blog to the Post didn't help
- using Sync/Async processing didn't change anything
- the Post has many objects that are expected to be not null, yet aren't required to be set in constructor - this generates compilator warnings

### Page URL

https://learn.microsoft.com/en-us/ef/core/saving/related-data#adding-a-graph-of-new-entities

### Content source URL

https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/saving/related-data.md

### Document Version Independent Id

c7517e7d-f011-a24c-53da-913771913998

### Article author

@ajcvickers

**Comment by timdeschryver** [comment]
I tried to reproduce this, and I initially received the following error 

```
Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Content', table 'EFDocs.dbo.Posts'; column does not allow nulls. UPDATE fails.
```

This occurs because I created a new project with [Nullable reference types](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references) enabled. Reading the comment "this generates compilator warnings", I expect this is also the case for you.

To fix the example, either remove the `<Nullable>enable</Nullable>` config in your csproj file, or set the content of the posts.

```cs
using (var context = new BloggingContext())
{
    var blog = new Blog
    {
        Url = "http://blogs.msdn.com/dotnet",
        Posts = new List<Post>
        {
            new Post { Title = "Intro to C#", Content = string.Empty},
            new Post { Title = "Intro to VB.NET", Content = ""},
            new Post { Title = "Intro to F#", Content = "Intro to F#"},
        }
    };

    context.Blogs.Add(blog);
    context.SaveChanges();
}
```

If wanted, I can update the docs by assigning a value to the content properties.

**Comment by litoj** [comment]
The primary issue is rather that there is no clarification what happens with the `post.BlogId` - does it just magically figure out it is from the object above? Also that part in particular did not work for me - I wasn't able to save it like that and instead it always threw an error about breaking FK constraint - percisely because the posts didn't get the `BlogId` set.

**Comment by timdeschryver** [comment]
@litoj it's normal that the entity as id `0`. During the insert it should receive the appropriate id.
For example, with Sql Server the `Id` column will be the Primary Key, which is auto incremented (`[BlogId] [int] IDENTITY(1,1) NOT NULL`)

**Comment by litoj** [comment]
Yes, but how does the `post.BlogId` get set? Because in my tests it didn't and that's why I got the errors.

**Comment by timdeschryver** [comment]
Sorry @litoj , I don't know - normally, that should also work automatically.

**Comment by litoj** [comment]
I resolved to saving each entity manually and after getting the parent entity ID by saving it, I only then added the child collection to it and then it worked. In the end my actual code looks like this:
```csharp
// duplicate `entity` while duplicating all of its children

        await DbContext.TestTemplate.AddAsync(entity);
        var bundles = entity.Bundles;
        entity.Bundles = [];
        await DbContext.SaveChangesAsync();
        entity.Bundles = bundles;
        await DbContext.SaveChangesAsync();
```

**Comment by litoj** [comment]
In my previous comment, I actually am using this feature successfully, but only for the grandchildren (Bundles->Slots). It seems that it works fine for one layer of children, but doesn't cascade for multiple layers (Template->Bundles->Slots).
